packages/cli/lib/services/deploy.service.ts

```diff
@@ -132,7 +132,35 @@ class DeployService {
 
         const singleDeployMode = Boolean(optionalSyncName || optionalActionName);
 
-        const successfulCompile = await compileAllFiles({ fullPath, debug });
+        let successfulCompile = false;
+
+        if (singleDeployMode) {
+            const scriptName: string = String(optionalSyncName || optionalActionName);
+            const type = optionalSyncName ? 'syncs' : 'actions';
+            const providerConfigKey = response.parsed.integrations.find((integration) => {
+                if (optionalSyncName) {
+                    return integration.syncs.find((sync) => sync.name === scriptName);
+                } else {
+                    return integration.actions.find((action) => action.name === scriptName);
```

Comment:
you can get rid of the `if` by doing 
```
return integration[type].find((script) => script.name === scriptName)
```

Reply:
Typescript yelled at me when I tried to make it dynamic